### PR TITLE
tests 5,6 and 3 optional have done.

### PR DESCRIPTION
**Pull request by Allessyer**
Tests 5,6 and 3 optional have been implemented. Check it.

### CODE REVIEW

**Comment by rolkon** [comment]
Hello!
If you restart the Kernel and run the notebook again, an error is thrown in your test cases. This is because you use the "create_persons" function, which is not defined until later.

Move your code cell to the bottom.

File: notebooks/Assignment_Solution.ipynb [hunk 1]
```diff
@@ -277,6 +277,118 @@
...
+    "        self.persons[0].water  = self.persons[0].LOWEST_WATER_PCT_TO_SURVIVE * self.persons[0].weight - 1\n",
+    "        self.persons[0].day_actions()\n",
+    "        self.assertIsInstance(self.persons[0].state, Dead)\n",
+    "    \n",
+    "    #test_3_optional\n",
+    "    #A person shall have an antibody of the same type as his virus, \n",
+    "    #if his virus has obtained strength below zero.\n",
+    "    \n",
+    "    def test_night_actions_3_optional(self):\n",
+    "        \n",
```
Comment (rolkon) [inline notebooks/Assignment_Solution.ipynb:346]: For a more complete test, maybe also test here that the person does NOT have the antibodies before he/she gets infected.

File: notebooks/Assignment_Solution.ipynb [hunk 2]
```diff
@@ -277,6 +277,118 @@
...
+    "    def test_day_actions_6(self):\n",
+    "\n",
+    "        self.persons[0].get_infected(SARSCoV2()) #change person's temperature\n",
+    "        self.persons[0].set_state(SymptomaticSick(self.persons[0]))\n",
+    "        #case 1 if temperature is not ok\n",
+    "        self.persons[0].temperature = self.persons[0].MAX_TEMPERATURE_TO_SURVIVE + 1\n",
+    "        self.persons[0].day_actions()\n",
+    "        self.assertIsInstance(self.persons[0].state, Dead)\n",
+    "        #case 2 if water balance is not ok\n",
+    "        self.persons[0].get_infected(Cholera()) #change person's water balance\n",
```
Comment (rolkon) [inline notebooks/Assignment_Solution.ipynb:335]: Here, the state of the person should not have changed (still be Dead). the get_infected method of a dead person does absolutely nothing. I don't think an infection would have taken place here. Set the state of the person to healthy again, or choose another person in the array.

File: notebooks/Assignment_Solution.ipynb [hunk 3]
```diff
@@ -277,6 +277,118 @@
...
+    "        \n",
+    "    #test_5\n",
+    "    #A person shall change his health state from infected \n",
+    "    #(with symptoms) to healthy, if the virus has obtained strength below zero.\n",
+    "\n",
+    "    def test_night_actions_5(self):\n",
+    "        \n",
+    "        virus_type = SARSCoV2(strength=2.0)\n",
+    "        self.persons[0].get_infected(virus_type)\n",
+    "        self.persons[0].set_state(SymptomaticSick(self.persons[0]))\n",
```
[the remark this answers is not in the document]
Comment (Allessyer) [reply notebooks/Assignment_Solution.ipynb:314]: Yes, I directly test SymptomaticSick State.

File: notebooks/Assignment_Solution.ipynb [hunk 4]
```diff
@@ -277,6 +277,118 @@
...
+    "        vi = self.persons[0].virus.get_type()\n",
```
[the remark this answers is not in the document]
Comment (Allessyer) [reply notebooks/Assignment_Solution.ipynb:315]: Yes, it's a typo.

File: notebooks/Assignment_Solution.ipynb (hunk 2 above)
Comment (Allessyer) [reply notebooks/Assignment_Solution.ipynb:335]: Missed it, yes, thank you. Changed.

File: notebooks/Assignment_Solution.ipynb (hunk 1 above)
Comment (Allessyer) [reply notebooks/Assignment_Solution.ipynb:346]: It's get_infected() in Healthy State functionality to check does virus type in the antibody list or not before being sick.

Comment (rolkon) [reply notebooks/Assignment_Solution.ipynb]: I see, makes sense.

**Pull request merged**